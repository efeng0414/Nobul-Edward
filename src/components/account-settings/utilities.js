import React from "react";
import SettingsFieldItem from "../settings/common/settings-field-item";
import {
  FIRST_NAME,
  LAST_NAME,
  PHONE,
  ADDRESS1,
  ADDRESS2,
  COUNTRY,
  PROVINCE_OR_STATE,
  CITY,
  POSTAL_OR_ZIP_CODE,
  PASSWORD,
  EMAIL
} from "../../../core/api-transform/users";
import TextField from "../form-fields/text-field";
import PhoneField from "../form-fields/phone-field";
import PostalField from "../form-fields/postal-field";
import CountryField from "../form-fields/country-field";
import ProvinceField from "../form-fields/province-field";
import ChangePasswordForm from "../forms/common/change-password-form";
import ChangeEmailForm from "../forms/common/change-email-form";

const LOCALE_KEY_PREFIX = "consumerSettings.account.";

export const fieldsMap = {
  [FIRST_NAME]: {
    labelKey: "firstName",
    type: TextField,
    required: true
  },
  [LAST_NAME]: {
    labelKey: "lastName",
    type: TextField,
    required: true
  },
  [EMAIL]: {
    labelKey: "email",
    type: ChangeEmailForm,
    required: true
  },
  [PASSWORD]: {
    labelKey: "password",
    type: ChangePasswordForm,
    required: true
  },
  [PHONE]: {
    labelKey: "phoneNumber",
    type: PhoneField,
    required: true
  },
  [ADDRESS1]: {
    labelKey: "address1",
    type: TextField,
    halfWidth: true,
    required: true
  },
  [ADDRESS2]: {
    labelKey: "address2",
    type: TextField,
    halfWidth: true,
    required: false
  },
  [COUNTRY]: {
    labelKey: "country",
    type: CountryField,
    halfWidth: true,
    required: true
  },
  [PROVINCE_OR_STATE]: {
    labelKey: "province",
    type: ProvinceField,
    halfWidth: true,
    required: true
  },
  [CITY]: {
    labelKey: "city",
    type: TextField,
    halfWidth: true,
    required: true
  },
  [POSTAL_OR_ZIP_CODE]: {
    labelKey: "postcode",
    type: PostalField,
    halfWidth: true,
    required: true
  }
};

export const generateSettingFieldsItemForProfile = ({
  userProfile,
  onEditClick
}) => {
  return Object.entries(fieldsMap).map(([key, { labelKey, halfWidth }]) => {
    return (
      <SettingsFieldItem
        labelKey={`${LOCALE_KEY_PREFIX}${labelKey}`}
        value={userProfile[key]}
        onEditClick={onEditClick}
        key={key}
        id={key}
        halfWidth={halfWidth}
      />
    );
  });
};
