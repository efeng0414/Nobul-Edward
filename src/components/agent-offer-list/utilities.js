import { REBATE_COMMISSION } from "../../../core/api-transform/users";
import { REBATE_COMMISSION as OFFER_REBATE_COMMISSION } from "../../../core/api-transform/offers";
import { BUY, SELL } from "../../../core/api-transform/jobs";

export const getJobType = ({ jobDetails, jobTypeProp }) => {
  if (jobTypeProp) return jobTypeProp;
  return jobDetails[REBATE_COMMISSION] !== undefined ||
    jobDetails[OFFER_REBATE_COMMISSION] !== undefined
    ? BUY
    : SELL;
};
