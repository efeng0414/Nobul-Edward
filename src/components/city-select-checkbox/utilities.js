// Helper function to toggle on and off all neighborhoods of the polygon data.
// This is only used to turn on/off Toronto polygons at the moment.
export const toggleSelectedNeighborhoodsHelper = ({
  selectAll,
  mapPolygons,
  currentPolygons,
  setPolygons
}) => {
  const neighbourhoods = mapPolygons ? mapPolygons.neighbourhoods.features : [];

  if (selectAll) {
    const selectedPolygons = {};

    neighbourhoods.map(
      polygon =>
        (selectedPolygons[polygon.properties.polygonId] = {
          name: polygon.properties.name,
          provinceOrState: polygon.properties.provinceOrState,
          country: polygon.properties.country
        })
    );

    setPolygons({
      ...currentPolygons,
      ...selectedPolygons
    });
  } else {
    const selectedPolygons = { ...currentPolygons };
    neighbourhoods.map(
      polygon => delete selectedPolygons[polygon.properties.polygonId]
    );
    setPolygons(selectedPolygons);
  }
};

const neighborhoodsPolygonsMap = new WeakMap();

export const allNeighborhoodsSelected = ({ mapPolygons, currentPolygons }) => {
  if (!mapPolygons || !mapPolygons.neighbourhoods) {
    return false; // No neighborhoods to be selected.
  }

  let neighborhoodPolygonIds = mapPolygons
    ? neighborhoodsPolygonsMap.get(mapPolygons)
    : null;

  if (!neighborhoodPolygonIds) {
    const neighbourhoods = mapPolygons.neighbourhoods.features;
    neighborhoodPolygonIds = neighbourhoods.map(
      polygon => polygon.properties.polygonId
    );
    neighborhoodsPolygonsMap.set(mapPolygons);
  }

  const selectedPolygonIds = Object.keys(currentPolygons);
  if (selectedPolygonIds.length < neighborhoodPolygonIds.length) {
    return false; // If there are less polygons selected than neighborhoods, they sure arent all selected.
  }

  // Check if all neighborhood IDs are included in selectedPolygonIds
  return neighborhoodPolygonIds.every(
    elem => selectedPolygonIds.indexOf(elem) > -1
  );
};

// Check for Toronto, show/hide city button. Will change to generic in future.
export const showCityCheckbox = ({ placesApiResult }) => {
  return (
    placesApiResult.address_components &&
    placesApiResult.address_components[0].short_name === "Toronto"
  );
};
