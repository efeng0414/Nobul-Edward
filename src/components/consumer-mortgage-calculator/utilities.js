export const amortizationRange = [5, 10, 15, 20, 25, 30, 35];
