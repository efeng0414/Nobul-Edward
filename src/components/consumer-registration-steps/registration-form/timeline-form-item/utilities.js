export const optionArray = [
  { value: 1, key: "timeline1" },
  { value: 2, key: "timeline2" },
  { value: 3, key: "timeline3" },
  { value: 4, key: "timeline4" },
  { value: 5, key: "timeline5" }
];
