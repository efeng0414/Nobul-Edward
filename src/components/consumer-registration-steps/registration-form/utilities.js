import { BUY, SELL, BOTH } from "../../../../core/constants/shared";

export const fieldMap = {
  LOOKING_FOR: "lookingFor",
  TIMELINE: "timeline",
  PRE_APPROVED_MORTGAGE: "preApprovedMortgage",
  AREA_OF_INTEREST: "areaOfInterest",
  TERM: "term",
  CONSENT: "consent",
  MORTGAGE_AMOUNT: "mortgageAmount",
  CONTRACT_TO_BROKERAGE: "contractToBrokerage",
  CONSENT_FOR_NEWS: "consent_for_news",
  CONSENT_FOR_EMAIL: "email_notifications"
};

export const getFormPayload = ({ signUpValues, formValues }) => {
  const postData = {
    ...signUpValues,
    ...formValues
  };
  postData.mortgageAmount = !postData.preApprovedMortgage
    ? "0"
    : postData.mortgageAmount.replace(/^0+/, "");

  postData.consentForNews = postData[fieldMap.CONSENT].includes(
    fieldMap.CONSENT_FOR_NEWS
  );

  postData.allowEmailNotifications = postData[fieldMap.CONSENT].includes(
    fieldMap.CONSENT_FOR_EMAIL
  );

  postData[fieldMap.LOOKING_FOR] = postData[fieldMap.LOOKING_FOR].includes(BOTH)
    ? [BUY, SELL]
    : [postData[fieldMap.LOOKING_FOR]];

  return postData;
};
