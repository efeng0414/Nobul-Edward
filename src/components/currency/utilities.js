export const CURRENCY_SYMBOLS = {
  CA: "<span class='currency__code'>CA</span>$",
  US: "<span class='currency__code'>US</span>$",
  EU: "€",
  GB: "£"
};
