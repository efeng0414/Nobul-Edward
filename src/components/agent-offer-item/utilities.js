import React from "react";
import { Icon, Tag } from "antd";
import { JOB_ACCEPTED, JOB_DELETED } from "../../../core/constants/jobs";
import {
  OFFER_OPEN,
  OFFER_ACCEPTED,
  OFFER_REJECTED,
  OFFER_WITHDRAWN,
  OFFER_PENDING_VERIFICATION
} from "../../../core/constants/offers";
import { translate } from "../../utilities/locale";

export const TAG_MAP = {
  ACCEPTED: {
    condition: ({ correspondingJob, offerDetail, withdraw }) =>
      !withdraw &&
      correspondingJob.status === JOB_ACCEPTED &&
      offerDetail.status === OFFER_ACCEPTED,
    jsx: ({ intl }) => (
      <Icon type="check" key="ACCEPTED">
        {translate(intl, "offerAccepted")}
      </Icon>
    )
  },
  OPEN: {
    condition: ({ correspondingJob, offerDetail, withdraw }) =>
      !withdraw && offerDetail.status === OFFER_OPEN,
    jsx: ({ intl }) => (
      <Tag className="agent-offer__tag agent-offer__tag--open" key="OPEN">
        {translate(intl, "offerOpen")}
      </Tag>
    )
  },
  REJECTED: {
    condition: ({ correspondingJob, offerDetail, withdraw }) =>
      !withdraw && offerDetail.status === OFFER_REJECTED,
    jsx: ({ intl }) => (
      <Tag
        className="agent-offer__tag agent-offer__tag--rejected"
        key="REJECTED"
      >
        {translate(intl, "offerRejected")}
      </Tag>
    )
  },
  WITHDRAWN: {
    condition: ({ correspondingJob, offerDetail, withdraw }) =>
      withdraw || offerDetail.status === OFFER_WITHDRAWN,
    jsx: ({ intl }) => (
      <Tag
        className="agent-offer__tag agent-offer__tag--withdrawn"
        key="WITHDRAWN"
      >
        {translate(intl, "offerWithdrawn")}
      </Tag>
    )
  },

  JOB_DELETED: {
    condition: ({ correspondingJob, offerDetail, withdraw }) =>
      !withdraw && correspondingJob.status === JOB_DELETED,
    jsx: ({ intl }) => (
      <Tag
        className="agent-offer__tag agent-offer__tag--job-deleted"
        key="JOB_DELETED"
      >
        {translate(intl, "offer.jobDeletedTag")}
      </Tag>
    )
  },

  PENDING_VERIFICATION: {
    condition: ({ correspondingJob, offerDetail, withdraw }) =>
      !withdraw && offerDetail.status === OFFER_PENDING_VERIFICATION,
    jsx: ({ intl }) => (
      <Tag
        className="agent-offer__tag agent-offer__tag--pending-verification"
        key="OPEN"
      >
        {translate(intl, "offerPendingVerification")}
      </Tag>
    )
  }
};
