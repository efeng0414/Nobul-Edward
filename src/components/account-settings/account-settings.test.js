import React from "react";
import renderer from "react-test-renderer";
import { IntlProvider } from "react-intl";
import { fieldsMap } from "./utilities";
import { translate } from "../../utilities/locale";
import {
  COUNTRY,
  FIRST_NAME,
  PASSWORD
} from "../../../core/api-transform/users";

import AccountSettingsWrapped from "./component";
import { AGENTS, CONSUMERS } from "../../../core/api-transform";
import {
  AGENT_USER_TYPE,
  CONSUMER_USER_TYPE
} from "../../../core/constants/users";

describe("Account Settings", () => {
  const {
    WrappedComponent: { WrappedComponent: AccountSettings }
  } = AccountSettingsWrapped;

  const componentProps = {
    intl: global.getIntl(),
    userProfile: {},
    form: {
      getFieldDecorator: config => jsx => jsx,
      setFieldsValue: function(values) {
        this.setFieldValueCalledWith = values;
      },
      validateFieldsAndScroll: callback => {
        callback();
      },
      resetFields: () => {},
      setFieldValueCalledWith: {}
    },
    authUserId: "userId",
    userType: "consumer",
    consentForNews: false,
    consentForEmailNotifications: false,
    updateProfile: () => {},
    updatePassword: () => {},
    updateConsentForNews: jest.fn(),
    updateConsentForEmailNotifications: jest.fn(),
    updateEmail: () => {},
    isEmailUnique: () => {}
  };

  const componentJSX = <AccountSettings {...componentProps} />;

  describe("Component Snapshot", () => {
    const snapshotJSX = (
      <IntlProvider {...global.intl}>{componentJSX}</IntlProvider>
    );

    const componentSnapshot = renderer.create(snapshotJSX).toJSON();
    it("Should match stored snapshot", () => {
      expect(componentSnapshot).toMatchSnapshot();
    });

    describe("Component mount", () => {
      const functionKeysToReset = ["toggleModal"];
      const componentPrototypeResets = functionKeysToReset.reduce(
        global.createComponentPrototypeResets({
          component: AccountSettings
        }),
        {}
      );
      const componentMount = global.Enzyme.mount(
        componentJSX,
        global.getIntlContext({ intl: componentProps.intl })
      );

      const componentInstance = componentMount.instance();

      beforeEach(() => {
        functionKeysToReset.map(
          global.resetPropertyComponentProperty({
            component: AccountSettings,
            componentMount: componentMount,
            componentPrototypeResets
          })
        );
      });

      describe("toggleModal", () => {
        it("Should toggle the boolean value of isModalOpen", () => {
          const isModalOpen = false;
          const currentState = { isModalOpen };
          const newState = { isModalOpen: !isModalOpen };

          componentInstance.setState(currentState);
          componentInstance.toggleModal();
          expect(componentMount.state()).toEqual(newState);
        });

        it("Should call the resetFields function with the keys of the fieldsMap object", () => {
          const mockResetFields = jest.fn();
          componentProps.form.resetFields = mockResetFields;
          componentMount.setProps(componentProps);
          componentInstance.toggleModal();
          expect(mockResetFields).toBeCalledWith(Object.keys(fieldsMap));
        });
      });

      describe("onEditClick", () => {
        const mockEvent = {
          id: "eventId",
          target: {
            getAttribute: jest.fn()
          }
        };
        const testOnEditClick = fieldKey => {
          const InputToRender = fieldsMap[fieldKey].type;
          mockEvent.target.getAttribute.mockReturnValue(fieldKey);
          const getAttrSpy = jest.spyOn(mockEvent.target, "getAttribute");
          const toggleModalSpy = jest.spyOn(componentInstance, "toggleModal");
          componentInstance.onEditClick(mockEvent);
          expect(getAttrSpy).toHaveBeenCalledWith("id");

          const test = componentInstance.renderModalField();
          const renderModalField = JSON.stringify(test);

          const modalFieldComponent = JSON.stringify(
            <InputToRender
              form={componentProps.form}
              initialValue={componentProps.userProfile[fieldKey]}
              label={translate(
                componentProps.intl,
                fieldsMap[fieldKey].labelKey
              )}
              name={fieldKey}
              country={componentProps.userProfile[COUNTRY]}
              updatePassword={componentProps.updatePassword}
              onDoneClick={componentInstance.toggleModal}
              checkIfEmailUnique
              required
            />
          );
          expect(renderModalField).toEqual(modalFieldComponent);
          expect(componentInstance.activeFieldKey).toEqual(fieldKey);
          expect(toggleModalSpy).toBeCalled();
        };
        it("Should set the field key to the event's target id", () => {
          testOnEditClick(FIRST_NAME);
        });
      });

      describe("getUserTypeNode", () => {
        it("Should return the correct user type for agents", () => {
          const agentPropUserType = AGENT_USER_TYPE;
          componentMount.setProps({ userType: agentPropUserType });
          expect(componentInstance.getUserTypeNode()).toEqual(AGENTS);
        });

        it("Should return the correct user type for consumer", () => {
          const agentPropUserType = CONSUMER_USER_TYPE;
          componentMount.setProps({ userType: agentPropUserType });
          expect(componentInstance.getUserTypeNode()).toEqual(CONSUMERS);
        });
      });

      describe("shouldHideFooter", () => {
        const passwordFieldKey = PASSWORD;
        const notPasswordFieldKey = "username";

        it("Should return true if activeKey is password", () => {
          componentInstance.activeFieldKey = passwordFieldKey;
          expect(componentInstance.shouldHideFooter()).toEqual(true);
        });

        it("Should return false if activeKey is not password", () => {
          componentInstance.activeFieldKey = notPasswordFieldKey;
          expect(componentInstance.shouldHideFooter()).toEqual(false);
        });
      });

      describe("toggleConsentForNews", () => {
        it("Should call updateConsentforNews with correct data", () => {
          const expectedArg = {
            consentForNews: !componentInstance.props.consentForNews,
            consumerId: componentInstance.props.authUserId
          };

          componentInstance.toggleConsentForNews();
          expect(
            componentInstance.props.updateConsentForNews
          ).toHaveBeenCalledWith(expectedArg);
        });
      });

      describe("toggleConsentForEmailNotificatoins", () => {
        it("Should call updateConsentForEmailNotifications with correct data", () => {
          const expectedArg = {
            consumerId: componentInstance.props.authUserId,
            emailNotifications: !componentInstance.props
              .consentForEmailNotifications
          };

          componentInstance.toggleConsentForEmailNotifications();
          expect(
            componentInstance.props.updateConsentForEmailNotifications
          ).toHaveBeenCalledWith(expectedArg);
        });
      });
    });
  });
});
