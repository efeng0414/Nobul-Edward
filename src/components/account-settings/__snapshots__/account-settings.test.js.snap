// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Account Settings Component Snapshot Should match stored snapshot 1`] = `
<div
  className="consumer-account-settings"
>
  <div
    className="field-items"
  >
    <div
      className="settings-field-item"
    >
      <label
        className="label"
      >
        First Name
        <i
          className="anticon anticon-edit"
          id="firstName"
          onClick={[Function]}
        />
      </label>
    </div>
    <div
      className="settings-field-item"
    >
      <label
        className="label"
      >
        Last Name
        <i
          className="anticon anticon-edit"
          id="lastName"
          onClick={[Function]}
        />
      </label>
    </div>
    <div
      className="settings-field-item"
    >
      <label
        className="label"
      >
        Email
        <i
          className="anticon anticon-edit"
          id="email"
          onClick={[Function]}
        />
      </label>
    </div>
    <div
      className="settings-field-item"
    >
      <label
        className="label"
      >
        Change Password
        <i
          className="anticon anticon-edit"
          id="password"
          onClick={[Function]}
        />
      </label>
      <p
        className="value"
      >
        ********
      </p>
    </div>
    <div
      className="settings-field-item"
    >
      <label
        className="label"
      >
        Phone Number
        <i
          className="anticon anticon-edit"
          id="phone"
          onClick={[Function]}
        />
      </label>
    </div>
    <div
      className="settings-field-item half-width"
    >
      <label
        className="label"
      >
        Address Line 1
        <i
          className="anticon anticon-edit"
          id="address1"
          onClick={[Function]}
        />
      </label>
    </div>
    <div
      className="settings-field-item half-width"
    >
      <label
        className="label"
      >
        Address Line 2
        <i
          className="anticon anticon-edit"
          id="address2"
          onClick={[Function]}
        />
      </label>
    </div>
    <div
      className="settings-field-item half-width"
    >
      <label
        className="label"
      >
        Country
        <i
          className="anticon anticon-edit"
          id="country"
          onClick={[Function]}
        />
      </label>
    </div>
    <div
      className="settings-field-item half-width"
    >
      <label
        className="label"
      >
        Province/State
        <i
          className="anticon anticon-edit"
          id="provinceOrState"
          onClick={[Function]}
        />
      </label>
    </div>
    <div
      className="settings-field-item half-width"
    >
      <label
        className="label"
      >
        City
        <i
          className="anticon anticon-edit"
          id="city"
          onClick={[Function]}
        />
      </label>
    </div>
    <div
      className="settings-field-item half-width"
    >
      <label
        className="label"
      >
        Postal/Zip Code
        <i
          className="anticon anticon-edit"
          id="postalOrZipCode"
          onClick={[Function]}
        />
      </label>
    </div>
    <div
      className="settings-field-item switch-field-item"
    >
      <span
        checked={false}
        className="switch ant-switch"
        defaultChecked={false}
        onChange={[Function]}
        onClick={[Function]}
        onKeyDown={[Function]}
        onMouseUp={[Function]}
        tabIndex={0}
      >
        <span
          className="ant-switch-inner"
        />
      </span>
      <div
        className="settings-field-item-container"
      >
        <p
          className="value"
        >
          Promotions / Newsletters
        </p>
      </div>
    </div>
  </div>
</div>
`;
